app/src/main/java/com/danosoftware/galaxyforce/services/googleplay/GooglePlayServices.java

```diff
@@ -0,0 +1,225 @@
+package com.danosoftware.galaxyforce.services.googleplay;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.drive.Drive;
+import com.google.android.gms.games.Games;
+import com.google.android.gms.games.GamesClient;
+import com.google.android.gms.games.SnapshotsClient;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+import static com.danosoftware.galaxyforce.constants.GameConstants.RC_SIGN_IN;
+
+public class GooglePlayServices {
+
+    /* logger tag */
+    private static final String ACTIVITY_TAG = "GooglePlayServices";
+
+    private enum ConnectedState {
+        NO_ATTEMPT, CONNECTED, DISCONNECTED;
+    }
+
+    private final Activity mActivity;
+    private final GoogleSignInClient signInClient;
+    private final GoogleSignInOptions signInOptions;
+    private ConnectedState connectedState;
+
+    public GooglePlayServices(Activity activity) {
+        Log.d(ACTIVITY_TAG, "Creating Billing client.");
+        this.mActivity = activity;
+        this.signInOptions =
+                new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_GAMES_SIGN_IN)
+                        // Add the APPFOLDER scope for Snapshot support.
+                        .requestScopes(Drive.SCOPE_APPFOLDER)
+                        // below scopes are replace the above deprecated scope
+//                        .requestScopes(new Scope(DriveScopes.DRIVE_APPDATA))
+//                        .requestScopes(new Scope(DriveScopes.DRIVE_FILE))
+                        .build();
+        this.signInClient = GoogleSignIn.getClient(
+                activity,
+                signInOptions);
```

Comment:
Might be better to only create client prior to making an API call rather than keeping an instance alive for the duration of the application. Maybe via a helper method for consistency.